app/lib/calendrical_events/provider/past_calendrical_events_page_controller.dart

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) 2023 Sharezone UG (haftungsbeschränkt)
+// Licensed under the EUPL-1.2-or-later.
+//
+// You may obtain a copy of the Licence at:
+// https://joinup.ec.europa.eu/software/page/eupl
+//
+// SPDX-License-Identifier: EUPL-1.2
+
+import 'dart:async';
+
+import 'package:date/date.dart';
+import 'package:flutter/material.dart';
+import 'package:group_domain_models/group_domain_models.dart';
+import 'package:rxdart/rxdart.dart';
+import 'package:sharezone/calendrical_events/analytics/past_calendrical_events_page_analytics.dart';
+import 'package:sharezone/calendrical_events/models/calendrical_event.dart';
+import 'package:sharezone/sharezone_plus/subscription_service/subscription_service.dart';
+import 'package:sharezone/timetable/src/widgets/events/event_view.dart';
+import 'package:sharezone/util/api/course_gateway.dart';
+import 'package:sharezone/util/api/school_class_gateway.dart';
+import 'package:sharezone/util/api/timetable_gateway.dart';
+
+class PastCalendricalEventsPageController extends ChangeNotifier {
+  late PastCalendricalEventsPageState state;
+  EventsSortingOrder _sortingOrder = EventsSortingOrder.descending;
+  EventsSortingOrder get sortingOrder => _sortingOrder;
```

Comment:
Feels a bit weird to have `sortingOrder` state here but also a `state` property.  
But also I have the feeling that doing it otherwise (include `PastCalendricalEventsPageState .sortingOrder`) might make it more complex to handle changing the sorting while e.g. loading/having an error. I'm not sure.  

So either:
1. Rename `state`, my idea would be `.eventsState` 
2. Add the `sortingOrder` to each of the `PastCalendricalEventsPageState` and remove from the controller.